sandbox/main.qml

```diff
@@ -59,6 +58,11 @@ StatusWindow {
         }
     }
 
+    readonly property real scaleRatio: Math.min(rootWindow.width / Theme.screenWidth, rootWindow.height / Theme.screenHeight)
+    onScaleRatioChanged: {
+        Theme.scaleFactor = scaleRatio;
+    }
+
```

Comment:
Shouldn't this be done in the `Theme` itself? You still have access to the attached `Window` and `Screen` properties